crates/cairo-1-hint-processor/src/hint_processor.rs

```diff
@@ -13,14 +14,33 @@ use cairo_rs::{
 };
 use felt::Felt252;
 use num_integer::Integer;
+use num_traits::cast::ToPrimitive;
 use num_traits::identities::Zero;
 use std::{collections::HashMap, ops::Mul};
 
-use crate::dict_manager::DictManagerExecScope;
-
 /// HintProcessor for Cairo 1 compiler hints.
 struct Cairo1HintProcessor {}
 
+/// Extracts a parameter assumed to be a buffer.
+fn extract_buffer(buffer: &ResOperand) -> (&CellRef, Felt252) {
+    let (cell, base_offset) = match buffer {
+        ResOperand::Deref(cell) => (cell, 0.into()),
+        ResOperand::BinOp(BinOpOperand {
+            op: Operation::Add,
+            a,
+            b,
+        }) => (
+            a,
+            extract_matches!(b, DerefOrImmediate::Immediate)
+                .clone()
+                .value
+                .into(),
+        ),
+        _ => panic!("Illegal argument for a buffer."),
```

Comment:
Why a panic?

Reply:
not sure actually, this code has been taken from cairo repository without modification.